src/cmd/bootstrap.ts

```diff
@@ -113,29 +113,31 @@ export const processValues = async (
 ): Promise<Record<string, any> | undefined> => {
   const { ENV_DIR, VALUES_INPUT } = deps.env()
   let originalValues: Record<string, any> | undefined
-  let derivedValues: Record<string, any>
   if (deps.isChart) {
     deps.debug.log(`Loading chart values from ${VALUES_INPUT}`)
     originalValues = deps.loadYaml(VALUES_INPUT) as Record<string, any>
     const storedSecrets = await deps.getStoredClusterSecrets()
     if (storedSecrets) originalValues = merge(originalValues, storedSecrets)
     await deps.writeValues(originalValues, true)
-    derivedValues = (await deps.hfValues()) as Record<string, any>
   } else {
     deps.debug.log(`Loading repo values from ${ENV_DIR}`)
     // we can only read values from ENV_DIR if we can determine cluster.providers
     if (deps.loadYaml(`${ENV_DIR}/env/cluster.yaml`, { noError: true })?.cluster?.provider) {
       originalValues = (await deps.hfValues()) as Record<string, any>
     }
-    derivedValues = originalValues as Record<string, any>
-  }
-  // do we need to create a custom CA?
-  if (derivedValues?.charts && derivedValues.charts['cert-manager']?.issuer === 'custom-ca') {
-    await deps.createCustomCA(derivedValues)
   }
   // generate secrets that don't exist yet
-  const generatedSecrets = await deps.generateSecrets(originalValues)
-  await deps.writeValues(generatedSecrets, false)
+  let generatedSecrets = await deps.generateSecrets(originalValues)
+
+  // do we need to create a custom CA? if so add it to the secrets
+  const cm = get(originalValues, 'charts.cert-manager', {})
+  if (cm.issuer === 'custom-ca' || cm.issuer === undefined) {
+    if (cm.customRootCA && cm.customRootCAKey) {
+      deps.debug.info('Skipping custom RootCA generation')
+    } else generatedSecrets = deps.createCustomCA(originalValues as Record<string, any>)
+  }
+  // we have generated all we need, now store the values and merge in the secrets
+  await deps.writeValues(merge(originalValues, generatedSecrets), true)
```

Comment:
why do you want to overwrite values? it can lead to overwriting already generated secrets